controller/cache/cache.go

```diff
@@ -437,7 +437,7 @@ func (c *liveStateCache) getCluster(server string) (clustercache.ClusterCache, e
 		return nil, fmt.Errorf("error getting cluster: %w", err)
 	}
 
-	if !c.canHandleCluster(cluster) {
+	if !c.canHandleCluster(cluster) && c.clusterSharding == nil {
```

Comment:
I think the nil check should come first isn't it ?

Reply:
Hi @anandf, I have updated the code to handle nil ptr dereference separately